src/util/router.js

```diff
@@ -79,6 +79,49 @@ export const forceReplace = ({ router, unsavedChanges }, location) => {
   return router.replace(location);
 };
 
+// Returns a scrollBehavior function to pass to createRouter().
+export const createScrollBehavior = (animation = 'smooth') => {
+  // Function to cancel a previous attempt to scroll
+  let cancel;
+  const scrollTo = (scrollId) => {
+    const el = document.querySelector(`[data-scroll-id="${scrollId}"]`);
+    return el != null ? { el, top: 10, behavior: animation } : null;
+  };
+  return (to, from, savedPosition) => {
+    if (cancel != null) cancel();
+    if (to.hash === '') return to.path === from.path ? savedPosition : null;
+    const scrollId = to.hash.replace('#', '');
+    // Certain characters would lead to an invalid CSS selector above, so we
+    // limit the set of accepted characters.
+    if (!/^[\w-]+$/.test(scrollId)) return null;
+    // Try to scroll right away. If the element doesn't exist, then we need to
+    // wait for it. In that case, we will attempt to scroll on an interval,
+    // giving up after 3 minutes or so.
+    return scrollTo(scrollId) ?? new Promise(resolve => {
+      let tries = 0;
+      const intervalId = setInterval(
```

Comment:
can we do something to avoid `interval`?

An idea: raise a custom event on the `<body>` when Entity feed is `mounted` and listen to that event here.

Reply:
Good idea, let me see if I can avoid it somehow. That may simplify testing as well.